base/tuple.jl

```diff
@@ -123,6 +123,23 @@ end
 _ntuple(f, n) = (@_noinline_meta; ([f(i) for i = 1:n]...))
 
 # inferrable ntuple
+ntuple{F}(f::F, ::Type{Val{0}}) = (@_inline_meta; ())
```

Comment:
Cheers re. using `where` syntax. (Though IIRC, method specialization on function arguments happens automagically when the function arguments are called directly in the method's body, so chances are type parameter `F` is unnecessary.) Best!